rclc/src/rclc/executor.c

```diff
@@ -1833,7 +1833,7 @@ rclc_executor_spin_some(rclc_executor_t * executor, const uint64_t timeout_ns)
         if (rc == RCL_RET_OK) {
           RCUTILS_LOG_DEBUG_NAMED(
             ROS_PACKAGE_NAME,
-            "Subscription added to wait_set_subscription[%ld]",
+            "Subscription added to wait_set_subscription[%d]",
```

Comment:
`index` is `size_t`, so most appropriate way is to use `%zu`?

```suggestion
            "Subscription added to wait_set_subscription[%zu]",
```

Reply:
Here is the compiler warning, so I use %d and it fixes the warning, %zu should be fine too. Let me know if you insist to use %zu.
/b/mros/installer_no_patches/microros_ws/firmware/mcu_ws/uros/rclc/rclc/src/rclc/executor.c: In function 'rclc_executor_spin_some':
/b/mros/installer_no_patches/microros_ws/firmware/mcu_ws/uros/rclc/rclc/src/rclc/executor.c:1836:13: warning: format '%ld' expects argument of type 'long int', but argument 5 has type 'size_t' {aka 'unsigned int'} [-Wformat=]
 1836 |             "Subscription added to wait_set_subscription[%ld] ",
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1837 |             executor->handles[i].index);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                 |
      |                                 size_t {aka unsigned int}
/b/mros/installer_no_patches/microros_ws/firmware/mcu_ws/uros/rclc/rclc/src/rclc/executor.c:1836:60: note: format string is defined here
 1836 |             "Subscription added to wait_set_subscription[%ld] ",
      |                                                          ~~^
      |                                                            |
      |                                                            long int
      |                                                          %d

Reply:
FYI I remember finding some incompatibilities with zu in some ARM GCC toolchains, I am not sure, but it needs investigation

Reply:
```shell
format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=] 1836 | "Subscription added to wait_set_subscription[%d]", | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 1837 | executor->handles[i].index); | ~~~~~~~~~~~~~~~~~~~~~~~~~~ | | | size_t {aka long unsigned int}
```